src/core.js

```diff
@@ -300,15 +300,13 @@ jQuery.extend({
 	makeArray: function( arr, results ) {
 		var ret = results || [];
 
-		if ( arr != null ) {
-			if ( isArraylike( Object(arr) ) ) {
-				jQuery.merge( ret,
-					typeof arr === "string" ?
-					[ arr ] : arr
-				);
-			} else {
-				push.call( ret, arr );
-			}
+		if ( isArraylike( Object(arr) ) ) {
+			jQuery.merge( ret,
+				typeof arr === "string" ?
+				[ arr ] : arr
+			);
+		} else if ( arr != null ) {
+			push.call( ret, arr );
```

Comment:
@mzgol, so with the actual change we have:
```
   raw     gz Sizes
255363  75554 dist/jquery.js
 84483  29366 dist/jquery.min.js

   raw     gz Compared to master @ 8f13997e89ca325a49f7581fad7e85fe37bad166
   -33     +3 dist/jquery.js
    -3     +6 dist/jquery.min.js
```
After, if I re-add the additionnal `if` in the `makeArray` function we have:
```
   raw     gz Sizes
255379  75556 dist/jquery.js
 84485  29364 dist/jquery.min.js

   raw     gz Compared to master @ 8f13997e89ca325a49f7581fad7e85fe37bad166
   -17     +5 dist/jquery.js
    -1     +4 dist/jquery.min.js

   raw     gz Compared to each-undefined @ 2d1766f83adb5f9032e2645f5ac55268c566e75b
   +16     +2 dist/jquery.js
    +2     -2 dist/jquery.min.js

   raw     gz Compared to last run
   +16     +2 dist/jquery.js
    +2     -2 dist/jquery.min.js
```